assets/js/player.js

```diff
@@ -352,6 +352,7 @@ if (video_data.params.save_player_pos) {
     let lastUpdated = 0;
 
     if(!hasTimeParam) set_seconds_after_start(rememberedTime);
+    if (rememberedTime >= video_data.length_seconds - 20 && !hasTimeParam) set_seconds_after_start(0);
```

Comment:
I think that setting the start time twice can add some delay, as the player would have to seek (= load more data) twice.

You can prevent that by extracting the common condition like so:
```js
if(!hasTimeParam) {
  if (rememberedTime >= video_data.length_seconds - 20)
    set_seconds_after_start(0);
  else
    set_seconds_after_start(rememberedTime);
}

Reply:
Works well!